mlem/contrib/venv.py

```diff
@@ -0,0 +1,173 @@
+                os.getenv("VIRTUAL_ENV") is None
+                or sys.prefix == sys.base_prefix
+            ):
+                raise MlemError("No virtual environment detected.")
+            echo(EMOJI_PACK + f"Detected the virtual env {sys.prefix}")
+            env_dir = sys.prefix
+            env_exe = os.path.join(env_dir, "bin", "python")
+        else:
+            echo(EMOJI_PACK + f"Creating virtual env {self.target}...")
+            context = self.create_virtual_env()
+            env_dir = context.env_dir
+            os.environ["VIRTUAL_ENV"] = context.env_dir
+            env_exe = context.env_exe
+        echo(EMOJI_PACK + "Installing the required packages...")
+        # Based on recommendation given in https://pip.pypa.io/en/latest/user_guide/#using-pip-from-your-program
+        install_cmd = [env_exe, "-m", "pip", "install"]
+        if self.no_cache:
+            install_cmd.append("--no-cache-dir")
+        install_cmd.extend(obj.requirements.to_pip())
+        subprocess.run(install_cmd, check=True)
+        echo(
+            EMOJI_OK
+            + f"virtual environment `{self.target}` is ready, activate with `source {self.target}/bin/activate`"
+        )
+        return env_dir
+
+
+class CondaBuilder(EnvBuilder):
+
+    type: ClassVar = "conda"
+
+    python_version: str = "3"
+    """The python version to use"""
+    current_env: Optional[bool] = False
+    """Whether to install in the current conda env"""
+
+    def create_virtual_env(self):
+        ret = None
+        create_cmd = ["--prefix", self.target, f"python={self.python_version}"]
+        try:
+            ret = subprocess.run(
+                ["conda", "create", "-y", *create_cmd], check=True
+            )
+        except FileNotFoundError:
+            echo(
+                EMOJI_FAIL
+                + f"Couldn't create the conda environment `{self.target}`"
+            )
+        return ret.returncode
+
+    def get_installed_packages(self, env_dir):
+        return subprocess.check_output(["conda", "list", "--prefix", env_dir])
+
+    def build(self, obj: MlemModel):
+        pip_based_packages = obj.requirements.to_pip()
+        conda_based_packages = obj.requirements.to_conda()
+
+        if self.current_env:
+            conda_default_env = os.getenv("CONDA_DEFAULT_ENV", None)
+            if conda_default_env == "base" or conda_default_env is None:
+                raise MlemError("No conda environment detected.")
+            echo(EMOJI_PACK + f"Detected the conda env {sys.prefix}")
+            env_dir = sys.prefix
+            env_exe = sys.executable
+        else:
+            assert self.target is not None
+            ret = self.create_virtual_env()
+            assert ret == 0
+            env_dir = self.target
+            env_exe = os.path.join(env_dir, "bin", "python")
+        try:
+            if conda_based_packages:
+                ret = subprocess.run(
+                    [
+                        "conda",
```

Comment:
probably should check that `conda` cmd is available

Reply:
done